server/handler.go

```diff
@@ -220,52 +263,53 @@ func updateAgent(w http.ResponseWriter, r *http.Request) {
 
 func clientLogin(w http.ResponseWriter, r *http.Request) {
 	user := &database.Login{}
-	err := json.NewDecoder(r.Body).Decode(user)
+	if err := json.NewDecoder(r.Body).Decode(user); err != nil {
+		sendErrorMsg(w, err, http.StatusInternalServerError)
+		return
+	}
 
 	if err := database.ClientLogin(user.UserName, user.Pass); err != nil {
-		log.Error(err)
-		w.WriteHeader(http.StatusUnauthorized)
+		sendErrorMsg(w, err, http.StatusUnauthorized)
 		return
 	}
 
 	userID, err := database.GetUserID(user.UserName)
 	if err != nil {
-		log.Error(err)
-		w.WriteHeader(http.StatusInternalServerError)
+		sendErrorMsg(w, err, http.StatusInternalServerError)
 		return
 	}
 
 	if err := sessionSet(w, r, userID); err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		log.Error(err)
+		sendErrorMsg(w, err, http.StatusInternalServerError)
 		return
 	}
-	w.WriteHeader(http.StatusOK)
+	sendErrorMsg(w, nil, http.StatusOK)
 }
 
 func clientLogout(w http.ResponseWriter, r *http.Request) {
 	if err := closeSession(w, r); err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
+		sendErrorMsg(w, err, http.StatusInternalServerError)
 		return
 	}
-	w.WriteHeader(http.StatusOK)
+	sendErrorMsg(w, nil, http.StatusOK)
```

Comment:
sendErrorMsg(w, nil, http.StatusOK) - Really ? 

Either rename it to sendResponse, or create other to send OK

Reply:
I know it and I`m going to change it after Oleksander's pull request merging.